lib/model/CIndividualModel.cc

```diff
@@ -339,6 +339,20 @@ void CIndividualModel::doPersistModelsState(core::CStatePersistInserter& inserte
     }
 }
 
+bool CIndividualModel::shouldPersistModel() const {
+    // If a feature model's memory usage is zero, it indicates that the model
+    // has been pruned and replaced with a stub. If all feature models are
+    // such stubs then we choose to not persist the model
+    // in order to reclaim memory after a persist/restore cycle.
+    std::size_t modelMemory{0};
+    for (const auto& feature : m_FeatureModels) {
+        for (const auto& model : feature.s_Models) {
+            modelMemory += model->memoryUsage();
+        }
+    }
+    return modelMemory > 0 ? true : false;
```

Comment:
Since the final condition is just `> 0` you can make this shortcircuit instead of checking all models.  This could be quite beneficial when there are huge numbers of models that _haven't_ been pruned.

```suggestion
    for (const auto& feature : m_FeatureModels) {
        for (const auto& model : feature.s_Models) {
            if (model->memoryUsage() > 0) {
                return true;
            }
        }
    }
    return false;
```